sirepo/template/ml.py

```diff
@@ -32,23 +36,36 @@
     validateFile='validate.npy',
 )
 
-
 def background_percent_complete(report, run_dir, is_running):
+    def _classifier_from_files():
+        f = PKDict(
+            decisionTree=_OUTPUT_FILE.dtClassifierConfusionFile,
+            knn=_OUTPUT_FILE.knnConfusionFile,
+        )
+        for k in f:
+           if run_dir.join(f[k]).exists():
+               return k
+        raise AssertionError(f'no output files generated for report={report}')
+
+    data = simulation_db.read_json(run_dir.join(template_common.INPUT_BASE_NAME))
     res = PKDict(
         percentComplete=0,
         frameCount=0,
     )
     if report == 'classificationAnimation' and not is_running:
         return PKDict(
-            percentComplete=100,
+            # TODO(e-carlin): is this the best I can do? Existing sims will have
+            # to be re-run because this is a cached value in supervisor db and
+            # won't be present in old sims
+            framesForClassifier=_classifier_from_files(),
```

Comment:
@moellep what do you think of this? I've added it because when a user has reports shown of the classifier page and then changes the classifier the reports were being retrieved for the new classifier and coming back with `report not generated` (because the output files are different based on the classifier selected). With this change users will have to re-run their simulations before being able to see any reports (framesForClassifier is "cached" in the supervisor db and prior to this commit it isn't present so the ng-if in the template will be false for both classifiers). Can you think of a better way to solve this problem?

Reply:
To be more specific, users who had existing output prior to this change will have to run their simulations again. After this change they won't. Knn takes just a few seconds to run (with the datasets I've used it can probably take a while with large datasets) so I figure it isn't that big of deal but lmk.

Reply:
Do we need to cached this in the job structure? Wouldn't be better to store it on disk and just request it as a report?

Reply:
To do that we would need to request this report then use it's output to decide if we should request the other reports. I don't immediately now how I would do that (or if we do that anywhere else). Are you aware of any other places where we do that right now? My understanding of the reports is that they are somewhat isolated from the rest of the code and there aren't good hooks to use their results for further business logic (but my understanding of the GUI is limited). Obviously that could change if we need it to.

Thinking about this more we could just change the logic in the html template, if we have frames but no framesForClassifier then assume knn since that is the only classifier implemented previously. That would get around the problem of users having to run again.

I think you are getting at a larger issue of putting "random" information in background_percent_complete as a way to communicate with the GUI. Maybe we could think about getting rid of the pattern all together...

Reply:
#2624 addresses the problem. I think there is a problem with putting random information in background_percent_complete: it's a hack. The hack then gets stored in the supervisor. It's bad style and confusing.

Reply:
I agree. How should we proceed? Should I try to implement #2624?

Reply:
We're in a bit of a trap. #2624 is an idea. Getting there is going to be tricky. I don't have a particular answer. How big is the data _classifier_from_files?

Reply:
It is very small. Just the name of the classifier ("dt", "knn", "svn", etc). It gets used [in the html template](https://github.com/radiasoft/sirepo/pull/2876/files#diff-3351e056502223a9461e52dd70e439bbR13) to decide which reports to request/display.

Reply:
Just leave it then. Not solvable right now

Reply:
You have the selected classifier already in the in.json which is read at the beginning of background_percent_complete() to get the state when the simulation was started. No need to check for output files  using _classifier_from_files(). I checked in a fix to the branch for this.